contracts/ToucanRegenBridge.sol

```diff
@@ -85,10 +85,23 @@ contract ToucanRegenBridge is Pausable, AccessControl {
     //      Constructor
     // ----------------------------------------
 
-    constructor(INCTPool nctPool_) {
+    constructor(
+        INCTPool nctPool_,
+        bytes32[] memory roles,
+        address[] memory accounts
+    ) {
+        require(roles.length > 0, "roles array must not be empty");
```

Comment:
Any particular reason for this requirement?  If so, please include it in the commit message.

But I can imagine situations where it's useful to deploy and then set up the roles later, e.g. if we hadn't yet decided who should get which roles, or if EOAs or multisigs still needed to be set up.

Reply:
Wouldn't the `for` loop fail if it tried to iterate over an array of length 0?

Reply:
I don't see why it would - what do you mean by fail?  The termination condition is `i < accounts.length`, so if `accounts.length` is `0` then it will just terminate immediately, which is what we want.

Reply:
I'm 99.9% sure but you should make sure yourself along the "trust but verify" maxim of  blockchain ;-)

The way to check is to write test cases for constructing with no roles and with one or more roles.

Reply:
You were correct. I don't know why I had this idea in my mind that it would fail. I'll remove the line requiring the `length > 0`. It might make sense to add the test though.